crates/cext/src/transpiler/passes/vf2.rs

```diff
@@ -38,28 +41,27 @@ pub struct VF2LayoutResult(Option<HashMap<VirtualQubit, PhysicalQubit>>);
 #[cfg(feature = "cbinding")]
 pub unsafe extern "C" fn qk_vf2_layout_result_has_match(layout: *const VF2LayoutResult) -> bool {
     let layout = unsafe { const_ptr_as_ref(layout) };
-    layout.0.is_some()
+    layout.0 != Vf2PassReturn::NoSolution
 }
 
 /// @ingroup QkVF2LayoutResult
-/// Get the number of virtual qubits in the layout.
+/// Check whether the result is an improvement to the trivial layout.
 ///
 /// @param layout a pointer to the layout
 ///
-/// @returns The number of virtual qubits in the layout
+/// @returns ``true`` if the VF2-based layout pass found an improved match.
 ///
 /// # Safety
 ///
 /// Behavior is undefined if ``layout`` is not a valid, non-null pointer to a
-/// ``QkVF2LayoutResult``. The result must have a layout found.
+/// ``QkVF2LayoutResult``.
 #[no_mangle]
 #[cfg(feature = "cbinding")]
-pub unsafe extern "C" fn qk_vf2_layout_result_num_qubits(layout: *const VF2LayoutResult) -> u32 {
```

Comment:
Is this function no longer needed? I see it was used in the tests, but is this also something users might want to check directly?

Reply:
It's not generally knowable by the `struct` in the case of `NoImprovement` - the struct doesn't store anything at all in those cases and will just return the identity mapping when asked for any qubit.  Neither Rust nor Python actually use a length of the corresponding object; you're just expect to know what the valid inputs are because you created it.